app/src/main/java/com/google/android/soundchecker/utils/ui/Waveform.kt

```diff
@@ -0,0 +1,72 @@
+package com.google.android.soundchecker.utils.ui
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment.Companion.Center
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun Waveform(
+    modifier : Modifier,
+    points: FloatArray?,
+) {
+    if (points == null) {
+        return;
+    }
+    Box(
+        modifier = modifier
+            .background(Color.White)
+            .padding(horizontal = 8.dp, vertical = 12.dp),
+        contentAlignment = Center
+    ) {
+        Canvas(
+            modifier = Modifier.fillMaxSize(),
```

Comment:
Does this erase the passed in modifier?

Reply:
The modifier is used for the outer box instead of the canvas inside. This allows users of the compose function to declare its size.